lib/net.js

```diff
@@ -425,12 +421,14 @@ Socket.prototype._destroy = function(exception, cb) {
   var self = this;
 
   function fireErrorCallbacks() {
+    var hadSeenError = self._errorEmitted;
+    self._errorEmitted = self._errorEmitted || Boolean(exception);
     if (cb) cb(exception);
-    if (exception && !self._writableState.errorEmitted) {
+    else if (exception && !hadSeenError) {
```

Comment:
Ah, since it was a single ExpressionStatement I preserved the braceless `if`. Fixing.